MF-454 - Use message Time field as a time for InfluxDB points

dborovcanin opened the pull request:
Signed-off-by: Dusan Borovcanin <dusan.borovcanin@mainflux.com>

### What does this do?
Use time from received message to set time value in InfluxDB.  

### Which issue(s) does this PR fix/relate to?
This pull request closes #454.

### Notes
This pull request slightly modifies the way messages are saved in InfluxDB. Field "Time" is removed and replaced by default "time" column of InfluxDB. Previously, "time" kept information about timestamp when the message is saved in the database, so that info is no longer persistent in InfluxDB. However, this information can be found in service logs if needed. This also means that the size of tag set of the single point is slightly reduced.

dborovcanin committed:
Use message time as Point time in InfluxDB

Signed-off-by: Dusan Borovcanin <dusan.borovcanin@mainflux.com>

drasko reviewed (changes requested):
As discussed in the issue - tagset is wrong.

This tagset will form huge number of different series. Only `channelID` and eventually `content-type` should be in a tagset.

dborovcanin committed:
Use actual message time

Update all Reader and Writer services to use time from the message
instead of time given from the corrseponding Writer service.

Signed-off-by: Dusan Borovcanin <dusan.borovcanin@mainflux.com>

dborovcanin committed:
Remove message time check

Messages time less than 2**28 represent time relative to the current time so Writer used to convert this to the correct value, i.e. msg.Time += time.Now(). However, this step is optional and should really be a part of the app on top of Mainflux or could be introduced with minor changes in Normalizer, Reader or Writer services, so there is no need for this to be supported out of box.

Signed-off-by: Dusan Borovcanin <dusan.borovcanin@mainflux.com>

drasko reviewed (dismissed):
LGTM

dborovcanin committed:
Use channel and publisher as tag keys

Move all the other Message fields to the field keys.

Signed-off-by: Dusan Borovcanin <dusan.borovcanin@mainflux.com>

drasko reviewed (approved):
LGTM
